pkg/controller/installation/typha_autoscaler_test.go

```diff
@@ -95,12 +103,44 @@ var _ = Describe("Test typha autoscaler ", func() {
 		Expect(err).To(BeNil())
 		verifyTyphaReplicas(c, 2)
 	})
+	It("should be degraded if there's not enough linux nodes", func() {
+		typhaMeta := metav1.ObjectMeta{
+			Name:      "calico-typha",
+			Namespace: "calico-system",
+		}
+		// Create a typha deployment
+		typha := &appsv1.Deployment{
+			TypeMeta:   metav1.TypeMeta{Kind: "Deployment", APIVersion: "apps/v1"},
+			ObjectMeta: typhaMeta,
+		}
+		err := c.Create(ctx, typha)
+		Expect(err).To(BeNil())
+
+		statusManager.On("SetDegraded", "Failed to autoscale typha", "not enough linux nodes to schedule typha pods on, require 4 and have 2")
+
+		// Create a few nodes
+		_ = createNode(c, "node1", map[string]string{"kubernetes.io/os": "linux"})
+		_ = createNode(c, "node2", map[string]string{"kubernetes.io/os": "linux"})
+		_ = createNode(c, "node3", map[string]string{"kubernetes.io/os": "window"})
+		_ = createNode(c, "node4", map[string]string{"kubernetes.io/os": "window"})
+
+		// Create the autoscaler and run it
+		ta := newTyphaAutoscaler(c, statusManager, typhaAutoscalerPeriod(10*time.Millisecond))
+		ta.start()
+
+		time.Sleep(1 * time.Second)
```

Comment:
Instead of using a sleep here it should work just to do `ta.isDegraded()`, that would ensure that autoscale was ran and SetDegraded would be set if degraded.

Reply:
Technically there's still a race there, although it's a bit smaller. The initial autoscale run is done in the go routine, so the degraded check could theoretically beat the run.

I think putting this to a milli second then doing the degraded check would be good, just to ensure this. WDYT?

Reply:
The ta.isDegraded() will block until the loop and select happens once and writes the degraded status back. Before the loop even runs we do an autoscale and set degraded so I don't think there is a race here.